fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingThrottle.java

```diff
@@ -325,6 +332,7 @@ public <R> CompletableFuture<R> buildCommitRetryAsync(@Nonnull final BiFunction<
             loadConfig();
             return common.getRunner().runAsync(context -> common.getRecordStoreBuilder().copyBuilder().setContext(context).openAsync().thenCompose(store -> {
                 expectedIndexStatesOrThrow(store, context);
+                booker.resetStoreTimerSnapshot();
```

Comment:
IIUC this will be off from the transaction. So, it could be something like:
1. getReadVersion
2. resetStoreTimerSnapshot
3. indexSome records
4. commit (successfully)
5. getReadVersion & open store -> times out

The metrics would include everything except the setup for the previous transaction, but not the current transaction.


If you did this instead in `booker.handleLimitsPostRunnerTransaction` when there's success, and before calling `AsyncUtil.whileTrue`, I think it would have the timer for the entire duration of the transaction. 

All of this is an indication that we should stop using the runner within index building, but that's a much bigger change.